File: output/ChatGPT/avatar/Java/C++/codeforces_400_A.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool isSolution(int columnsCount, string x) {
    vector<bool> allXs(columnsCount, true);
    
    for (int i = 0; i < 12; i++) {
        if (x[i] != 'X') {
            allXs[i % columnsCount] = false;
        }
    }
    
    for (int i = 0; i < columnsCount; i++) {
        if (allXs[i]) {
            return true;
        }
    }
    
    return false;
}

int main() {
    int t;
    cin >> t;
    
    for (int i = 0; i < t; i++) {
        int possibleSolutions = 0;
        string s;
        cin >> s;
        string solutions = "";
        
        if (isSolution(12, s)) {
            solutions += " 1x12";
            possibleSolutions++;
        }
        
        if (isSolution(6, s)) {
            solutions += " 2x6";
            possibleSolutions++;
        }
        
        if (isSolution(4, s)) {
            solutions += " 3x4";
            possibleSolutions++;
        }
        
        if (isSolution(3, s)) {
            solutions += " 4x3";
            possibleSolutions++;
        }
        
        if (isSolution(2, s)) {
            solutions += " 6x2";
            possibleSolutions++;
        }
        
        if (isSolution(1, s)) {
            solutions += " 12x1";
            possibleSolutions++;
        }
        
        cout << possibleSolutions << solutions << endl;
    }
    
    return 0;
}

File: output/ChatGPT/avatar/Java/Python/codeforces_400_A.py
def is_solution(columns_count, x):
    all_Xs = [True] * columns_count
    for i in range(12):
        if x[i] != 'X':
            all_Xs[i % columns_count] = False
    for i in range(columns_count):
        if all_Xs[i]:
            return True
    return False

t = int(input())
for _ in range(t):
    possible_solutions = 0
    solutions = []
    s = input()
    
    if is_solution(12, s):
        solutions.append("1x12")
        possible_solutions += 1
    if is_solution(6, s):
        solutions.append("2x6")
        possible_solutions += 1
    if is_solution(4, s):
        solutions.append("3x4")
        possible_solutions += 1
    if is_solution(3, s):
        solutions.append("4x3")
        possible_solutions += 1
    if is_solution(2, s):
        solutions.append("6x2")
        possible_solutions += 1
    if is_solution(1, s):
        solutions.append("12x1")
        possible_solutions += 1

    print(possible_solutions, *solutions)